packages/sommelier.rb

```diff
@@ -47,246 +39,281 @@ class Sommelier < Package
+            xdpyinfo -display "${DISPLAY}" &>/dev/null
+          }
+          if ! checksommelierwayland || ! checksommelierxwayland ; then
+            [ -f  #{CREW_PREFIX}/bin/stopbroadway ] && stopbroadway
+            #{CREW_PREFIX}/sbin/sommelierd &> /dev/null &
+          fi
+          wait=3
+          until checksommelierwayland && checksommelierxwayland; do
+            [ "${wait}" -le "0" ] && break
+            (( wait = wait - 1 ))
+            sleep 3
+          done
+        
+          SOMMWPIDS="$(pgrep -f "sommelier.elf --parent" 2> /dev/null)"
+          SOMMWPROCS="$(pgrep -fa "sommelier.elf --parent" 2> /dev/null)"
+          SOMMXPIDS="$(pgrep -f "sommelier.elf -X" 2> /dev/null)"
+          SOMMXPROCS="$(pgrep -fa "sommelier.elf -X" 2> /dev/null)"
+        
+          if checksommelierwayland && checksommelierxwayland ; then
+            echo -e "sommelier processes running: ${SOMMWPIDS} ${SOMMXPIDS}"
+          else
+            echo "some sommelier processes failed to start"
+            echo -e "sommelier processes running: ${SOMMWPROCS} \\n ${SOMMXPROCS}"
+            exit 1
+          fi
+        EOF
+
         # stopsommelier
-        system "cat <<'EOF'> stopsommelier
-#!/bin/bash
-SOMM=\$(pgrep -fc sommelier.elf 2> /dev/null)
-if [[ \"$SOMM\" -gt \"0\" ]]; then
-  pkill -f sommelier.elf &>/dev/null
-  pkill -F #{CREW_PREFIX}/var/run/sommelier-wayland.pid &>/dev/null
-  pkill -F #{CREW_PREFIX}/var/run/sommelier-xwayland.pid &>/dev/null
-else
-  exit 0
-fi
-if [[ \"$(pgrep -fc sommelier.elf 2> /dev/null)\" -gt \"0\" ]]; then
-  echo \"sommelier failed to stop\"
-  exit 1
-else
-  echo \"sommelier stopped\"
-fi
-EOF"
+        IO.write 'stopsommelier', <<~EOF
+          #!/bin/bash
+          SOMM="$(pgrep -fc sommelier.elf 2> /dev/null)"
+          if [[ "${SOMM}" -gt "0" ]]; then
+            pkill -f sommelier.elf &>/dev/null
+            pkill -F #{CREW_PREFIX}/var/run/sommelier-wayland.pid &>/dev/null
+            pkill -F #{CREW_PREFIX}/var/run/sommelier-xwayland.pid &>/dev/null
+          else
+            exit 0
+          fi
+          if [[ "$(pgrep -fc sommelier.elf 2> /dev/null)" -gt "0" ]]; then
+            echo "sommelier failed to stop"
+            exit 1
+          else
+            echo "sommelier stopped"
+          fi
+        EOF
+        
         # restartsommelier
-        system "echo '#!/bin/bash' > restartsommelier"
-        system "echo 'stopsommelier && startsommelier' >> restartsommelier"
+        IO.write 'restartsommelier', <<~EOF
+          #!/bin/bash
+          stopsommelier && startsommelier
+        EOF
       end
     end
   end
 
   def self.install
     Dir.chdir('vm_tools/sommelier') do
-      system "DESTDIR=#{CREW_DEST_DIR} ninja -C builddir install"
+      system "DESTDIR=#{CREW_DEST_DIR} samu -C builddir install"
       Dir.chdir('builddir') do
+        FileUtils.mkdir_p ["#{CREW_DEST_PREFIX}/bin", "#{CREW_DEST_PREFIX}/sbin", "#{CREW_DEST_PREFIX}/etc", CREW_DEST_HOME]
```

Comment:
This also works and is slightly shorter:
`FileUtils.mkdir_p %W[#{CREW_DEST_PREFIX}/bin #{CREW_DEST_PREFIX}/sbin #{CREW_DEST_PREFIX}/etc CREW_DEST_HOME]`